src/drive/web/modules/actions/useActions.spec.jsx

```diff
@@ -0,0 +1,422 @@
+    it('is only displayed on mobile with a single file selected', () => {
+      const qualifyAction = getAction('qualify')
+
+      expect(
+        qualifyAction.displayCondition([{ type: 'file', id: 'abc' }])
+      ).toBe(true)
+
+      expect(
+        qualifyAction.displayCondition([
+          { type: 'file', id: 'abc' },
+          { type: 'file', id: 'def' }
+        ])
+      ).toBe(false)
+      expect(
+        qualifyAction.displayCondition([{ type: 'folder', id: 'abc' }])
+      ).toBe(false)
+    })
+
+    it('shows the qualification modal when activated', () => {
+      const qualifyAction = getAction('qualify')
+      const mockDocument = { id: 'abc' }
+      qualifyAction.action([mockDocument])
+      const actuallyCalledModal =
+        mockModalContextValue.pushModal.mock.calls[0][0]
+
+      expect(mockModalContextValue.pushModal).toHaveBeenCalledWith(
+        <EditDocumentQualification
+          document={mockDocument}
+          onQualified={actuallyCalledModal.props.onQualified}
+          onClose={actuallyCalledModal.props.onClose}
+        />
+      )
+    })
+  })
+
+  describe('versions action', () => {
+    it('is only displayed on mobile with a single file selected', () => {
+      const versionsAction = getAction('history')
+
+      expect(
+        versionsAction.displayCondition([{ type: 'file', id: 'abc' }])
+      ).toBe(true)
+
+      expect(
+        versionsAction.displayCondition([
+          { type: 'file', id: 'abc' },
+          { type: 'file', id: 'def' }
+        ])
+      ).toBe(false)
+      expect(
+        versionsAction.displayCondition([{ type: 'folder', id: 'abc' }])
+      ).toBe(false)
+    })
+
+    it('redirects to the history view when activated', () => {
+      const versionsAction = getAction('history')
+      const mockDocument = { id: 'abc' }
+      versionsAction.action([mockDocument])
+      expect(mockRouterContextValue.router.push).toHaveBeenCalledWith(
+        '/folder/file/abc/revision'
+      )
+      mockRouterContextValue.location.pathname = '/folder/123'
+      versionsAction.action([mockDocument])
+      expect(mockRouterContextValue.router.push).toHaveBeenCalledWith(
+        '/folder/123/file/abc/revision'
+      )
+    })
+  })
+
+  describe('offline action', () => {
+    it('is only displayed on mobile with a single file selected', () => {
+      const offlineAction = getAction('phone-download')
+
+      const validSelection = [{ type: 'file', id: 'abc' }]
+
+      global.__TARGET__ = 'mobile'
+      expect(offlineAction.displayCondition(validSelection)).toBe(true)
+
+      global.__TARGET__ = 'desktop'
+      expect(offlineAction.displayCondition(validSelection)).toBe(false)
+      expect(
+        offlineAction.displayCondition([
+          { type: 'file', id: 'abc' },
+          { type: 'file', id: 'def' }
+        ])
+      ).toBe(false)
+      expect(
+        offlineAction.displayCondition([{ type: 'folder', id: 'abc' }])
+      ).toBe(false)
+    })
+  })
+})
```

Comment:
I think that's good that all the action are now tested on several situations. However, I think it's difficult to get an overall picture of what's going on, situation per situation.

I think it could be good that at some point we make a test plan with:

- sets of known files / folders with checks
- personas (read/write desktop, read/write mobile, read desktop, read mobile)

And matrix test all different situations and check the displayed actions.

Then we could have a test like this.

```
const { displayedActions } = setup(fileSet1, {
   selected: ['file1', 'file2'],
   platform: 'android'
})
expect(displayedActions.toMatch(['share', 'openwith']))
```

- I find it easier to think in terms of situation than action per action.
- I think it helps when the test case can be read by a non technical person

Reply:
I think that's a good idea. It was easier to write them as a list like this, but we can definitely keep the structure you suggest in mind for a future improvement.